controllers/resize_pvc.go

```diff
@@ -79,27 +91,66 @@ func (r *MySQLClusterReconciler) findAndResizePVC(ctx context.Context, cluster *
 
 	selector, err := metav1.LabelSelectorAsSelector(sts.Spec.Selector)
 	if err != nil {
-		return fmt.Errorf("failed to parse selector: %w", err)
+		return nil, fmt.Errorf("failed to parse selector: %w", err)
 	}
 
 	var pvcs corev1.PersistentVolumeClaimList
 	if err := r.Client.List(ctx, &pvcs, client.MatchingLabelsSelector{Selector: selector}); err != nil {
-		return fmt.Errorf("failed to list PVCs: %w", err)
+		return nil, fmt.Errorf("failed to list PVCs: %w", err)
 	}
 
+	patches := make(map[types.NamespacedName]*unstructured.Unstructured)
+
 	for _, pvc := range pvcs.Items {
 		newSize, ok := pvcsToKeep[pvc.Name]
 		if !ok {
 			continue
 		}
 
-		pvc.Spec.Resources.Requests[corev1.ResourceStorage] = *newSize
+		supported, err := r.isVolumeExpansionSupported(ctx, &pvc)
+		if err != nil {
+			return nil, fmt.Errorf("failed to check if volume expansion is supported: %w", err)
+		}
+		if !supported {
+			log.Info("StorageClass used by PVC does not support volume expansion, skipped", "storageClassName", *pvc.Spec.StorageClassName, "pvcName", pvc.Name)
+			continue
+		}
+
+		pvcac := corev1ac.PersistentVolumeClaim(pvc.Name, pvc.Namespace).
+			WithSpec(corev1ac.PersistentVolumeClaimSpec().
+				WithResources(corev1ac.ResourceRequirements().
+					WithRequests(corev1.ResourceList{corev1.ResourceStorage: *newSize}),
```

Comment:
Is there possibility that the current size is already bigger than the new size?
For example, with the help of pvc-autoresizer.

Reply:
Sorry for the late reply, I have created a design document to discuss the specifications
Please review 🙏 
https://github.com/cybozu-go/moco/pull/412

Reply:
@d-kuro Thank you for the document. LGTM.